src/components/MiniPaper.vue

```diff
@@ -21,44 +21,37 @@ export default {
   watch: {
     isOpen(isOpen) {
       if (isOpen) {
-        this.scaleMiniMapToFitContent();
+        this.miniMapManager.scaleMiniMap();
       }
     },
   },
-  methods: {
-    scaleMiniMapToFitContent() {
-      setTimeout(() => {
-        this.miniPaper.scaleContentToFit({ padding: 10, maxScaleX: 0.5, maxScaleY: 0.5 });
-      });
+  computed: {
+    visibilityClass() {
+      return this.isOpen ? 'visible' : 'invisible';
+    },
+    miniMap() {
+      return this.miniMapManager.miniMapPaper;
     },
+  },
+  methods: {
     movePaper({ offsetX, offsetY }) {
-      const { x, y } = this.miniPaper.paperToLocalPoint(offsetX, offsetY);
-      const { sx, sy } = this.paper.scale();
+      const { sx: scaleX, sy: scaleY } = this.paper.scale();
       const { clientWidth, clientHeight } = this.paper.el;
-
-      this.paper.translate(
-        (clientWidth / 2) - (x * sx),
-        (clientHeight / 2) - (y * sy),
-      );
+      const { newX, newY } = this.miniMapManager.calculateNewPaperPosition(offsetX, offsetY, scaleX, scaleY, clientWidth, clientHeight);
+      this.paper.translate(newX, newY);
     },
   },
   async mounted() {
     await this.$nextTick();
 
-    this.miniPaper = new dia.Paper({
-      el: this.$refs.miniPaper,
-      model: this.graph,
-      width: 300,
-      height: 200,
-      interactive: false,
-    });
+    this.miniMapManager = MiniMapManager.factory(this.graph, this.$refs.miniPaper);
 
-    this.paper.on('render:done', this.scaleMiniMapToFitContent);
+    this.paper.on('render:done', () => this.miniMapManager.scaleMiniMap());
+    window.ProcessMaker.EventBus.$on('modeler-change', () => this.miniMapManager.scaleMiniMap());
 
-    window.ProcessMaker.EventBus.$on('modeler-change', this.scaleMiniMapToFitContent);
   },
   beforeDestroy() {
-    this.paper.off('render:done', this.scaleMiniMapToFitContent);
+    this.paper.off('render:done', () => this.miniMapManager.scaleMiniMap());
```

Comment:
This use of `off` is based on https://backbonejs.org/#Events-off. Here, you're passing a new arrow function as the `callback` parameter, which will be different from the function passed to `this.paper.on`. Because of this, I'm not sure if JointJS/Backbone will be able to recognize that you're trying to remove the event listener for the `this.miniMapManager.scaleMiniMap` call.

Looking at https://backbonejs.org/#Events-on, it seems like you can pass a context for the event listener and then retain the reference to the function:
```js
this.paper.on('render:done', this.miniMapManager.scaleMiniMap, this);
// ...
this.paper.off('render:done', this.miniMapManager.scaleMiniMap, this);
```
I'm not even really sure we need to worry about removing this event listener though, since this component will mount once and stay loaded until the user leaves the page. But I think cleaning up event listeners like this is good practise.

Reply:
Let me try it and see if it works.

Reply:
All seems good, the change has been made.